Core/Src/main.c

```diff
@@ -116,35 +122,35 @@ int main(void)
   {
     /* USER CODE END WHILE */
 	  sendData(TxData);
-	  HAL_Delay(10);
-	  if (flag == 1 && (currentMillis - previousMillis > 20))
-	  {
-	    TxData[0] = 1;
-	    sendData(TxData);
-	    previousMillis = currentMillis;
-	    TxData[0] = 0;
-	  }
-	  if (flag == 2 && (currentMillis - previousMillis > 20))
-	  {
-	    TxData[0] = 2;
-	    sendData(TxData);
-	    previousMillis = currentMillis;
-	    TxData[0] = 0;
-	  }
-	 if (flag == 3 && (currentMillis - previousMillis > 20))
-	 {
-	    TxData[0] = 3;
-	    sendData(TxData);
-	    previousMillis = currentMillis;
-	    TxData[0] = 0;
-	  }
-	  if (flag == 4 && (currentMillis - previousMillis > 20))
-	  {
-	    TxData[0] = 4;
-	    sendData(TxData);
-	    previousMillis = currentMillis;
-	    TxData[0] = 0;
-	  }
+		  HAL_Delay(10);
+		  if (flag == 1)
+		  {
+		    TxData[0] = 1;
+		    sendData(TxData);
+		    TxData[0] = 0;
+		    flag = 0;
+		  }
+		  if (flag == 2)
+		  {
+		    TxData[0] = 2;
+		    sendData(TxData);
+		    TxData[0] = 0;
+		    flag = 0;
+		  }
+		 if (flag == 3)
+		 {
+		    TxData[0] = 3;
+		    sendData(TxData);
+		    TxData[0] = 0;
+		    flag = 0;
```

Comment:
What do these flag values mean? Maybe create an `enum` for each of the flag value?

Reply:
Yeah I can definitely make those more organized and clear. They keep track of what button was pressed but I understand that isn't obvious